OSDOCS-11169: making CSI optional for microshift (#80823)
openshift/openshift-docs: modules/microshift-stopping-lvms-csi-snapshot.adoc

// Module included in the following assemblies:
//
// * microshift_storage/microshift-storage-plugin-overview.adoc

:_mod-docs-content-type: PROCEDURE
[id="microshift-stopping-lvms-csi-snapshot_{context}"]
== Stopping LVMS and CSI volume snapshot deployment

> Workloads with attached volumes must be manually stopped, and those volumes must then be manually deleted by the user. Use the following procedure to prevent the deployment of LVMS CSI driver and CSI volume snapshot controller.

Reviewer: MicroShift treats the deployment of the CSI driver and the CSI snapshot implementations independently. As said in comments below, "deployment" is probably too overloaded to be useful here.

I would suggest something like:
```suggestion
Before disabling the CSI driver or CSI snapshot implementation pods, any user workloads with attached volumes must be manually stopped. Those volumes must then be manually deleted by the user. MicroShift will not attempt to delete CSI implementation pods. Instead, MicroShift must be configured not to run one or both of the CSI pods during the start-up process. Use the following procedures to disable installation of one or both CSI components, then restart MicroShift.
```

// Module included in the following assemblies:
//
// * microshift_storage/microshift-storage-plugin-overview.adoc

:_mod-docs-content-type: PROCEDURE
[id="microshift-stopping-lvms-csi-snapshot_{context}"]
== Stopping deployments that run CSI implementations

> 

Reviewer 2: My understanding is that removing the CSI makes the cluster no longer compliant with CNCF, so we need an IMPORTANT admonition in here.

Reviewer: MicroShift deploys dynamically provisioned storage by default, so it meets CNCF standards. I understand the importance of being CNCF compliant from a dev perspective. It seems weird to me that a customer would care if their particular deployment met this requirement as well.

Author: resolving as discussed

> User workloads with attached volumes must be manually stopped before disabling the CSI driver or CSI snapshot implementation pods. Use the following procedure to disable installation of one or both CSI components.

Reviewer 3: Make into an active sentence:

> Before you disable the CSI driver or CSI snapshot implementation pods, you must manually stop any user workloads that have attached volumes. Use the following procedure to disable installation of one or both CSI components.

Author: done

[NOTE]
====

> MicroShift will not delete CSI implementation pods. MicroShift must be configured to disable installation of one or both of the CSI pods during the start-up process.

Reviewer 3: Attribute for "MicroShift"? {microshift-short}?

Active voice:

> You must configure MicroShift to disable installation of one or both of the CSI pods during the startup process.

![Screenshot from 2024-09-04 09-15-28](https://github.com/user-attachments/assets/0434bf68-ebe4-43dd-8432-fcea3b846f01)

Author: done

====

.Procedure

. To disable installation of the CSI driver, enter the following `driver` value under the `storage` section of the {microshift-short} configuration file:
+
[source,yaml]
----
# ...
  storage
   driver:'' <1>
# ...
----
<1> Valid values are `none` or `lvms`.
+
[NOTE]
====
If the `driver` value is empty or null, {microshift-short} defaults to deploying LVMS.
====

. To disable installation of the CSI snapshot controller, enter the following `optionalCsiComponents` value under the `storage` section of the {microshift-short} configuration file:
+
[source,yaml]
----
# ...
  storage
   optionalCsiComponents:'' <1>
# ...
----
<1> Accepted values are `csi-snapshot-controller`, `csi-snapshot-webhook`, or `none`. `none` is mutually exclusive with all other values.
+
[NOTE]
====
If the `optionalCsiComponents` value is empty or null, {microshift-short} defaults to deploying snapshot-controller and snapshot-webhook.
====

> . To restart {microshift-short} after the `driver` and `optionalCsiComponents` fields are specified with supported values in the {microshift-short} config file, run the following command:

Reviewer 3: ```suggestion
. To restart {microshift-short} after the `driver` and `optionalCsiComponents` fields are specified with supported values in the {microshift-short} configuration file, run the following command:
```

Not sure how "config" holds up with translation to other languages.

Author: done

+
[source,terminal]
----
$ sudo systemctl restart microshift
----
+
[NOTE]
====

> {microshift-short} does not redeploy the disabled components after restart.

Reviewer 3: ```suggestion
{microshift-short} does not redeploy the disabled components after a restart operation.
```

Author: done
